Libraries/PeriphDrivers/Include/MAX78002/spi.h

```diff
@@ -103,62 +147,108 @@ typedef enum {
+                                    //    ts_mask[0] = Target Select Pin 0
+                                    //    ts_mask[1] = Target Select Pin 1
+                                    //    ts_mask[n] = Target Select Pin n
+    // DMA
+    bool use_dma;
+    mxc_dma_regs_t *dma;
+} mxc_spi_init_t;
 
-/**
- * @brief   The information required to perform a complete SPI transaction
- *
- * This structure is used by blocking, async, and DMA based transactions.
- * @note   "completeCB" only needs to be initialized for interrupt driven (Async) and DMA transactions.
- */
-struct _mxc_spi_req_t {
-    mxc_spi_regs_t *spi; ///<Point to SPI registers
-    int ssIdx; ///< Slave select line to use (Master only, ignored in slave mode)
-    int ssDeassert; ///< 1 - Deassert SS at end of transaction, 0 - leave SS asserted
-    uint8_t *txData; ///< Buffer containing transmit data. For character sizes
-    ///< < 8 bits, pad the MSB of each byte with zeros. For
-    ///< character sizes > 8 bits, use two bytes per character
-    ///< and pad the MSB of the upper byte with zeros
-    uint8_t *rxData; ///< Buffer to store received data For character sizes
-    ///< < 8 bits, pad the MSB of each byte with zeros. For
-    ///< character sizes > 8 bits, use two bytes per character
-    ///< and pad the MSB of the upper byte with zeros
-    uint32_t txLen; ///< Number of bytes to be sent from txData
-    uint32_t rxLen; ///< Number of bytes to be stored in rxData
-    uint32_t txCnt; ///< Number of bytes actually transmitted from txData
-    uint32_t rxCnt; ///< Number of bytes stored in rxData
-
-    spi_complete_cb_t completeCB; ///< Pointer to function called when transaction is complete
+// Suppport names for backwards compatibility.
+struct _mxc_spi_reva2_req_t {
+    mxc_spi_regs_t *spi;     // Pointer to SPI registers
+    union {
+        int deassert;
+        int ssDeassert;      // ssDeassert - deprecated name
+    };
+
+    union {
+        uint8_t *tx_buffer;
+        uint8_t *txData;     // txData - deprecated name
+    };
+
+    union {
+        uint8_t *rx_buffer;
+        uint8_t *rxData;     // rxData - deprecated name
+    };
+
+    union {
+        uint32_t tx_len;     // Number of bytes to be sent from txData
+        uint32_t txLen;      // txLen - deprecated name
+    };
+
+    union {
+        uint32_t rx_len;     // Number of bytes to be stored in rxData
+        uint32_t rxLen;      // rxLen - deprecated name
+    };
+
+    union {
+        uint32_t tx_cnt;     // Number of bytes actually transmitted from txData
+        uint32_t txCnt;      // txCnt - deprecated name
+    };
+
+    union {
+        uint32_t rx_cnt;     // Number of bytes stored in rxData
+        uint32_t rxCnt;      // rxCnt - deprecated name
+    };
+
+    uint16_t tx_dummy_value; // Value of dummy bytes to be sent
+
+    // Chip Select Options
+    mxc_spi_target_t *target_pins; // Contains index, pins, and polarity mode
```

Comment:
I figure this name is okay because there is no `mxc_spi_controller_t` field. Plus SPI v2 does not currently support setting the SPI peripheral as a target. And the `mxc_spi_type_t` field is for selecting between the controller and target.